src/TodoItem.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo, TodoForChange } from './types/Todo';
+
+interface Props {
+  todo: Todo;
+  removeTodo: (todoIduserId: number) => void;
+  loadingTodos: number[];
+  updateTodoInfo: (todoId: number, newInfo: TodoForChange) => void;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  removeTodo,
+  loadingTodos,
+  updateTodoInfo,
+}) => {
+  const [titleText, setTitleText] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const titleField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  }, [isOpen]);
+
+  const handleDoubleClick = () => {
+    setIsOpen(true);
+  };
+
+  const handleDelete = () => {
+    removeTodo(todo.id);
+  };
+
+  const handleToggleChange = async () => {
+    await updateTodoInfo(todo.id, { completed: !todo.completed });
+  };
+
+  const handleEditTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleText(event.target.value);
+  };
+
+  const saveChanges = () => {
+    const newInfo: Partial<Pick<Todo, 'title'>> = {
+      title: titleText,
+    };
+
+    updateTodoInfo(todo.id, newInfo);
+    setIsOpen(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!titleText) {
+      removeTodo(todo.id);
+    }
+
+    setTitleText(titleText);
+    setIsOpen(false);
+    saveChanges();
+  };
+
+  const onEscape = (event: React.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      setTitleText(titleText);
+      setIsOpen(false);
+      saveChanges();
+    }
+  };
```

Comment:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/39e4c9b7-ab65-4fb3-b7c5-376b330e624c)
Why do you save changes on `escape`?